tools/update-rule-type-definitions.js

```diff
@@ -0,0 +1,831 @@
+							tsruleDefinitionStart:
+								member.name.end - ruleNameLength,
+							tsruleDefinitionEnd: member.end,
+						};
+						ruleDefinitionTextPositions.typeEnd = member.end;
+						ruleDefinitionTextPositionsMap.set(
+							ruleId,
+							ruleDefinitionTextPositions,
+						);
+					}
+				}
+			}
+		}
+	}
+	return ruleDefinitionTextPositionsMap;
+}
+
+/**
+ * Resolves the schema by dereferencing any $ref pointers.
+ * @param {Object} schema The schema to resolve with 'definition' property.
+ * @returns {Object} The resolved schema without $ref pointers and 'definition' property.
+ */
+function resolveSchema(schema) {
+	const resolvedSchema = deref(schema);
+
+	if (resolvedSchema) {
+		// eslint-disable-next-line no-unused-vars -- We omit definitions as we have used its values as reference.
+		const { definitions, ...restSchema } = resolvedSchema;
+
+		return restSchema;
+	}
+
+	return schema;
+}
+
+/**
+ * Groups enum values by their keys.
+ * @param {Object} prop properties of the schema with type 'object'.
+ * @returns {Object} An object mapping enum keys to their corresponding property names.
+ */
+function getEnumGroups(prop) {
+	const enumGroups = {};
+	let keyGroup;
+	let valueGroup;
+
+	for (const [key, value] of Object.entries(prop)) {
+		if (Array.isArray(value.enum)) {
+			const enumKey = JSON.stringify(value.enum);
+			if (!enumGroups[enumKey]) {
+				enumGroups[enumKey] = [];
+			}
+			enumGroups[enumKey].push(key);
+		}
+	}
+
+	if (Object.keys(enumGroups).length !== 0) {
+		for (const [key, value] of Object.entries(enumGroups)) {
+			if (Array.isArray(value) && value.length > 1) {
+				keyGroup = value;
+				valueGroup = JSON.parse(key);
+			}
+		}
+	}
+
+	return { keyGroup, valueGroup };
+}
+
+/**
+ * Groups the properties that have equal values by their keys.
+ * @param {Object} prop properties of the schema with type 'object'.
+ * @returns {Object} An object mapping equal values to their corresponding property names.
+ */
+function getPropValueThatIsEqual(prop) {
+	const equalValues = [];
+	let keyValues;
+	let propertyValue;
+
+	for (const [key, value] of Object.entries(prop)) {
+		if (value.type === "object") {
+			const objectValue = JSON.stringify(value);
+
+			if (!equalValues[objectValue]) {
+				equalValues[objectValue] = [];
+			}
+			equalValues[objectValue].push(key);
+		}
+	}
+
+	if (Object.keys(equalValues).length !== 0) {
+		for (const [key, value] of Object.entries(equalValues)) {
+			if (Array.isArray(value) && value.length > 1) {
+				keyValues = JSON.parse(key);
+				propertyValue = JSON.parse(key).properties;
+			}
+		}
+	}
+
+	return { keyValues, propertyValue };
+}
+
+/**
+ * Creates a string of enum values for a given array of enum values.
+ * @param {Array} enumValue An array of enum values.
+ * @returns {string} A string of enum values separated by ' | '.
+ */
+function getArrayValues(enumValue) {
+	const enumValues = enumValue
+		.filter(value => typeof value === "string")
```

Comment:
Why do we remove all non-string values instead of keeping all enum members and only quoting the strings?